jbpm-designer-backend/src/test/java/org/jbpm/designer/bpmn2/impl/Bpmn2UnmarshallingTest.java

```diff
@@ -1759,6 +1760,36 @@ public void testInputOutputSetsOfIoSpecification() throws Exception {
 
     }
 
+    @Test
+    public void testDisableBpsimDisplayViaProfileSetting() throws Exception {
+        Definitions definitions = loader.loadProcessFromJson("userTask.json", "false", "false");
+        assertEquals(0, definitions.getRelationships().size());
+    }
+
+    @Test
+    public void testDisableBpsimDisplayViaSystemProperty() throws Exception {
+        System.setProperty(EditorHandler.BPSIM_DISPLAY, "false");
+        Definitions definitions = loader.loadProcessFromJson("userTask.json", "false", "true");
+        // should still be 0 even tho "true" is passed for bpsimDisplay
+        assertEquals(0, definitions.getRelationships().size());
+        System.clearProperty(EditorHandler.BPSIM_DISPLAY);
+    }
+
+    @Test
+    public void testEnableBpsimDisplayViaSystemProperty() throws Exception {
+        System.setProperty(EditorHandler.BPSIM_DISPLAY, "true");
+        Definitions definitions = loader.loadProcessFromJson("userTask.json", "false", "false");
```

Comment:
I think is ok as is because in tests we can "simulate" users having the default jbpm.xml (where bpsim display will be "true") and that via system property they can override that setting.
I have to add one more property like zorder and bpsimdisplay today for forms so will see if can simplify the method signatures not to have to take a bunch of true/false strings. Here I think its ok.